server/core/terraform/terraform_client.go

```diff
@@ -53,6 +57,12 @@ type Client interface {
 
 	// EnsureVersion makes sure that terraform version `v` is available to use
 	EnsureVersion(log logging.SimpleLogging, v *version.Version) error
+
+	// ListAvailableVersions returns all available version of Terraform, if available; otherwise this will return an empty list.
+	ListAvailableVersions(log logging.SimpleLogging) ([]string, error)
+
+	// DetectVersion Extracts required_version from Terraform configuration in the specified project directory. Returns nil if unable to determine the version.
+	DetectVersion(projectDirectory string, log logging.SimpleLogging) *version.Version
```

Comment:
How come in `DetectVersion` the `log` argument is the last argument but in `EnsureVersion` the `log` argument is the first argument? Can we make these consistent?

```suggestion
	DetectVersion(log logging.SimpleLogging, projectDirectory string) *version.Version
```

Reply:
This was addressed in https://github.com/runatlantis/atlantis/pull/2843/commits/588cc9109d87be7c3c8a673c74efc9ff266349b4.